espflash/src/image_format/esp8266.rs

```diff
@@ -39,19 +38,20 @@ impl<'a> Esp8266Format<'a> {
 
         // Common header
         let flash_mode = flash_mode.unwrap_or_default() as u8;
-        let flash_freq = flash_freq.unwrap_or_default();
-        let flash_size = flash_size.unwrap_or_default();
-        let flash_config =
-            encode_flash_size(flash_size)? + encode_flash_frequency(Chip::Esp8266, flash_freq)?;
         let segment_count = image.ram_segments(Chip::Esp8266).count() as u8;
 
-        let header = EspCommonHeader {
+        let mut header = ImageHeader {
```

Comment:
Maybe add a reference to ESP8266 file header: https://docs.espressif.com/projects/esptool/en/latest/esp8266/advanced-topics/firmware-image-format.html#file-header?